File: datasets/python_source-code/Multiple-B1/organic_10.py
# Binary Tree Traversal - organic submission 10
# Iterative in-order using a manual while-True/break loop and an explicit
# "pending" list instead of a variable literally named stack.

class Node:
    def __init__(self, item):
        self.item = item
        self.left = None
        self.right = None


def insert(root, item):
    if root is None:
        return Node(item)
    if item < root.item:
        root.left = insert(root.left, item)
    else:
        root.right = insert(root.right, item)
    return root


def inorder_walk(root):
    output = []
    pending = []
    walker = root
    while True:
        if walker is not None:
            pending.append(walker)
            walker = walker.left
            continue
        if not pending:
            break
        walker = pending.pop()
        output.append(walker.item)
        walker = walker.right
    return output


if __name__ == "__main__":
    root = None
    for i in [50, 30, 70, 20, 40, 60, 80]:
        root = insert(root, i)
    print("In-order (manual stack):", inorder_walk(root))

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_b1/organic_10.java
// Binary Tree Traversal - organic submission 10
// Iterative in-order using explicit while-true and stack.

import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

class Node {
    int value;
    Node left;
    Node right;

    public Node(int value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

class Solution {
    public static Node insert(Node root, int val) {
        if (root == null) {
            return new Node(val);
        }
        if (val < root.value) {
            root.left = insert(root.left, val);
        } else {
            root.right = insert(root.right, val);
        }
        return root;
    }

    public static List<Integer> inorderIterative(Node root) {
        List<Integer> res = new ArrayList<>();
        Stack<Node> s = new Stack<>();
        Node curr = root;
        while (true) {
            if (curr != null) {
                s.push(curr);
                curr = curr.left;
            } else if (!s.isEmpty()) {
                curr = s.pop();
                res.add(curr.value);
                curr = curr.right;
            } else {
                break;
            }
        }
        return res;
    }

    public static void main(String[] args) {
        Node root = null;
        for (int v : new int[]{50, 30, 70, 20, 40, 60, 80}) {
            root = insert(root, v);
        }
        System.out.println("In-order (iterative loop): " + inorderIterative(root));
    }
}
